src/itest/java/uk/gov/companieshouse/company/links/steps/CompanyChargesSteps.java

```diff
@@ -0,0 +1,153 @@
+    public void patchEdpointIsCalled(){
+        verify(1, getRequestedFor(urlEqualTo("/company/" + this.companyNumber + "/links")));
+        verify(1, patchRequestedFor(urlEqualTo("/company/" + this.companyNumber + "/links")));
+        wireMockServer.stop();
+    }
+
+    @Given("Company profile stubbed with charges present for {string}")
+    public void company_profile_exists_with_charges(String companyNumber) {
+        this.companyNumber = companyNumber;
+        configureWiremock();
+        setGetAndPatchStubsFor(loadFileForCoNumber("profile-with-charges-links.json", this.companyNumber));
+    }
+
+    private void configureWiremock() {
+        wireMockServer = new WireMockServer(Integer.parseInt(port));
+        wireMockServer.start();
+        configureFor("localhost", Integer.parseInt(port));
+    }
+
+    private void setGetAndPatchStubsFor(String response){
+        stubFor(
+            get(urlEqualTo("/company/" + this.companyNumber + "/links"))
+                .willReturn(aResponse()
+                    .withStatus(200)
+                    .withHeader("Content-Type", "application/json")
+                    .withBody(response)));
+
+        stubFor(
+            patch(urlEqualTo("/company/" + this.companyNumber + "/links"))
+                .withRequestBody(containing("/company/" + this.companyNumber + "/charges"))
+                .willReturn(aResponse()
+                    .withStatus(200)));
+    }
+
+    private ResourceChangedData createChargesMessage(String companyNumber) {
+        return createBaseMessage(companyNumber, RESOURCE_KIND_CHARGES);
+    }
+
+    private ResourceChangedData createBaseMessage(String companyNumber, String kind) {
+        EventRecord event = EventRecord.newBuilder()
+                .setType("changed")
+                .setPublishedAt("2022-02-22T10:51:30")
+                .setFieldsChanged(Arrays.asList("foo", "moo"))
+                .build();
+
+        return ResourceChangedData.newBuilder()
+                .setContextId("context_id")
+                .setResourceId(companyNumber)
+                .setResourceKind(kind)
+                .setResourceUri("/company/"+companyNumber+"/charges")
+                .setData("{ \"key\": \"value\" }")
+                .setEvent(event)
+                .build();
+    }
+
+    private String loadFileForCoNumber(String fileName, String companyNumber) {
+        try {
+            String templateText = FileUtils.readFileToString(ResourceUtils.getFile("classpath:stubs/"+fileName), StandardCharsets.UTF_8);
+            return String.format(templateText, companyNumber, companyNumber); // extra args are ignored
+        } catch (IOException e) {
+            throw new RuntimeException(String.format("Unable to locate file %s", fileName));
+        }
+    }
+
+    @Given("Company links consumer api service is running And Stubbed Company Profile for {string} API GET and PATCH")
```

Comment:
Cosmetic: Maybe move this to top, to keep all "Givens" at top, then "Whens" and then "Thens" maybe?

Reply:
Done